uaclient/apt.py

```diff
@@ -496,6 +496,16 @@ def get_installed_packages() -> List[str]:
     return [entry.split("/")[0] for entry in package_list]
 
 
+def get_installed_packages_with_version(
+    include_versions: bool = False,
+) -> List[str]:
+    filters = ["--showformat=${Package}\\n"]
+    if include_versions:
+        filters = []
+    out, _ = system.subp(["dpkg-query", "-W", *filters])
+    return out.splitlines()
```

Comment:
Renan discovered last week that `dpkg-query -W` can return packages that are not installed. He changed the `get_installed_packages` function to this https://github.com/canonical/ubuntu-advantage-client/pull/2238/files#diff-e0ee470947c8a3ab51ba84f5750d9e55511cc77d91556b1f54d07f076a690d41R493

We should change this function to use `apt list --installed` as well

Reply:
I think the way to go is:
- add a flag parameter like `show_version` to the `get_installed_packages` function
- change the way to parse the output based on this flag

But I think this was on this PR somehow before the new function showed up, feel free to disagree and disregard this idea.